Faster merge tracks

almostimplemented opened the pull request:
Closes #472 

Small and low priority performance enhancement to `mido.merge_tracks`. Skip redundant data checks. When you have a `MidiFile` that you want to iterate over, its messages should already have passed `mido.checks`. 

I made `merge_tracks` skip the checks by default, but since it is exposed to clients, maybe it should continue existing behavior by default and only change for internal invocations (like iterating over a `MidiFile`).

rdoursenaud reviewed (commented):
Thanks!
The approach is good yet I'm not sure about the implementation details.
I'm not a big fan of skipping checks by default in the `tracks.py` methods.
I must recognize I don't know what would be the consequences of bad data there.
Also, in `messages.py`, shouldn't we check if we get `overrides` regardless of the `skip_checks` status?

almostimplemented commented:
> I'm not a big fan of skipping checks by default in the `tracks.py` methods.

I think you are right -- I will change the default to enforce checks and only pass the skip flag internally.

> Also, in `messages.py`, shouldn't we check if we get `overrides` regardless of the `skip_checks` status?

Oof, duh! What an awful mistake. Fixed.

almostimplemented commented:
For reference, the `benchmark.py` script in #472 takes 0.913 seconds on `main` and 0.175 seconds on the PR branch.

rdoursenaud commented on mido/midifiles/tracks.py:
```diff
@@ -91,24 +91,33 @@ def fix_end_of_track(messages):
     The messages are returned in playback order with delta times
     as if they were all in one track.
+
+    By default, assumes the messages in tracks have already been
```
The docstring is now out of sync with the declaration.

rdoursenaud commented:
@almostimplemented Could you please review the test error and fix the merging conflicts. You may want to rebase upon the current main as I have fixed a bunch of GitHub workflow test errors. Thanks!

rdoursenaud commented:
@akx Yeah, I'm pondering moving to dedicated objects instead, like what CircuitPython's [adafruit_midi](https://docs.circuitpython.org/projects/midi/en/latest/api.html).

I'm gathering all insights for v2 in https://github.com/orgs/mido/discussions/540

almostimplemented commented:
Sorry for the very delayed reply. 

I suppose if you plan to remove the checks altogether, this particular change won't be necessary. But if you want to roll it in, I fixed the broken test and rebased.

rdoursenaud commented:
No problem, thank you for your contribution!

I'll merge your work since it's useful for mido 1.x users even if I get rid of it for v.2.
